src/Orchard.Web/Modules/Orchard.MediaLibrary/Services/XmlRpcHandler.cs

```diff
@@ -60,7 +60,7 @@ public class XmlRpcHandler : IXmlRpcHandler {
             UrlHelper url) {
 
             var user = _membershipService.ValidateUser(userName, password);
```

Comment:
Whenever you call ValidateUser, use a new variable, for instance validationResult, then keep the `user` variable to be `validationResult.User`. This will prevent all the `user.User` changes.